openwisp_users/admin.py

```diff
@@ -41,6 +42,24 @@ class OrganizationUserInline(admin.StackedInline):
 class EmailRequiredMixin(forms.ModelForm):
     email = forms.EmailField(label=_('Email'), max_length=254, required=True)
 
+    def _clean_email(self, email):
+        try:
+            EmailAddress.objects.get(email=email)
```

Comment:
I'm not entirely sure here, think about this scenario:

- the user has a secondary address already existing in the system
- he modifies his email in the user form to match the secondary address

The email already exists so the validation will fail.

Another note: counting instead of retrieving the whole model instance will be faster, let's use that.

Could you instead try performing the following 2 checks?

1. no other user has this email address, eg: `User.objects.filter(email=email).count() < 1`
2. no other email address record for other users exists, eg: `EmailAddress.objects.filter(email=email).exclude(user=user).count() < 0` (you should be able to find the `user_id` somewhere in `self` or `self.cleaned_data`

Reply:
Can't believe I missed that!